### ISSUE TITLE
Masonry layout doesn't work properly

### ISSUE THREAD

**Issue by mdsaban** [hoarder-app/hoarder#241]
- Masonry layout not working properly. I see lots of available space in one column, but extra items are still added to another column

![image](https://github.com/hoarder-app/hoarder/assets/45597394/0aca971f-ab25-4b15-afe4-41a257eaa263)

**Comment by mdsaban** [comment]
@MohamedBassem I'm planning to pick this up in my next PR and replace the existing masonry library with something better.

**Comment by MohamedBassem** [comment]
Yeah, I know it doesn't work that well. If you're up to fixing it, go ahead!
